src/core/Elsa.Core/Services/Workflows/WorkflowRunner.cs

```diff
@@ -381,12 +381,24 @@ private async Task CheckIfCompositeEventAsync(bool isComposite, bool scheduleVal
         {
             try
             {
-                return await activityOperation(activityExecutionContext, activity);
+                var result = await activityOperation(activityExecutionContext, activity);
+
+                if (result is FaultResult)
+                {
+                    var faultResult = (result as FaultResult);
+
+                    if (faultResult?.Exception != null)
+                        throw faultResult.Exception;
+
+                    throw new Exception(faultResult?.Message);
+                }
```

Comment:
Maybe simplify this a little bit? E.g.

```csharp
if (result is FaultResult faultResult)
{
    if (faultResult.Exception != null)
        throw faultResult.Exception;

    throw new Exception(faultResult?.Message);
}
````

Reply:
I've made this change, didn't know is could be used like this :D